lib/team.rb

```diff
@@ -30,8 +30,44 @@ def self.find_team_name_by_id(team_id)
     team_name
   end
 
+  def self.all
+    @@all
+  end
+
   def self.count_of_teams
     @@all.count
+  end  
+
+  def self.highest_scoring_visitor
+    generate_score_data.last[:team_name]
+  end
+
+  def self.lowest_scoring_visitor
+    generate_score_data.first[:team_name]
+  end
+
+  def self.generate_score_data
```

Comment:
Interesting way to do this! I like the remapping aspect of this but am curious how long this could take if we had even more games! Would be interesting to know if code like this really affects processing time or not.